### PR TITLE
added controller for ajax cart

### PR DESCRIPTION
**Pull request by SimplTanishq**

### CODE REVIEW

File: includes/admin/class-simpl-checkout-settings.php [hunk 1]
```diff
@@ -138,6 +142,7 @@ public static function is_simpl_enabled_for_admins() {
 	}
 
 	public static function store_url() {
+		return 'shop.1bill.in';
```
Comment (abhaymaniyar) [inline includes/admin/class-simpl-checkout-settings.php:145]: Was this hardcoded only for testing purpose? If yes, please remove. If not, please explain why.

**Review by abhaymaniyar — CHANGES_REQUESTED** [review]

File: includes/widget/buy-now-button.php [hunk 2]
```diff
@@ -13,6 +13,9 @@
   // hook for cart page
   add_action( $buttonPosition_cart, 'simpl_add_to_cart_btn');
 }
+if(WC_Simpl_Settings::can_display_in_mini_cart()){
```
Comment (Chhekur) [inline includes/widget/buy-now-button.php:16]: should this be can or should?

File: includes/widget/buy-now-button.php [hunk 3]
```diff
@@ -44,3 +46,23 @@ function load_widget_script(){
   $script_url = WC_Simpl_Settings::widget_script_url();
   echo '<script type="text/javascript" src=' .$script_url. '></script>';
 }
+
+function simpl_enable_cta_in_mini_cart() {
```
Comment (Chhekur) [inline includes/widget/buy-now-button.php:50]: this single method is doing a lot of things, can be divided into multiple small reusable methods.

File: includes/widget/buy-now-button.php [hunk 4]
```diff
@@ -44,3 +46,23 @@ function load_widget_script(){
...
+
+  if($is_simpl_cta_enabled) {
+    echo '<script>';
+    echo 'document.addEventListener("DOMContentLoaded", function() {';
+    echo '    var dataElement = document.createElement("div");';
+    echo '    dataElement.id = "simpl-custom-data";';
+    echo '    dataElement.setAttribute("data-is-mini-cart-cta-enabled", "true");';
+    echo '    document.head.appendChild(dataElement);';
+    echo '});';
+    echo '</script>';
```
Comment (Chhekur) [inline includes/widget/buy-now-button.php:66]:
```php
$button_script = <<<SCRIPT
<script>
document.addEventListener("DOMContentLoaded", function() {
  var dataElement = document.createElement("div");
  dataElement.id = "simpl-custom-data";
  dataElement.setAttribute("data-is-mini-cart-cta-enabled", "true");
  document.head.appendChild(dataElement);
});
</script>
SCRIPT;

echo $button_script;
```

File: includes/widget/set-plugin-config.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  $is_simpl_cta_enabled = WC_Simpl_Settings::is_simpl_button_enabled() || $is_simpl_enabled_for_admin || $is_simpl_pre_qa_env;
+
+	if($is_simpl_cta_enabled) {
+		$simpl_plugin_settings = WC_Simpl_Settings::simpl_get_all_latest_settings();
+
+		if(array_key_exists('merchant_client_secret', $simpl_plugin_settings)) {
+			unset($simpl_plugin_settings['merchant_client_secret']);
+		}
+
+		echo( '<script type="text/javascript">var SimplPluginConfig = ' . json_encode($simpl_plugin_settings) . '</script>' );
```
Comment (rithiksharma100) [inline includes/widget/set-plugin-config.php:19]: this is against the Wordpress guidelines for publishing our plugin, there is a native method to include scripts, we shouldn't do it using `echo`

**Review by rithiksharma100 — CHANGES_REQUESTED** [review]

File: includes/widget/buy-now-button.php (hunk 2 above)
Comment (SimplTanishq) [reply includes/widget/buy-now-button.php]: following previous patterns of naming, other setting functions are named similarly

**Review by Chhekur — DISMISSED** [review]

**Review by rithiksharma100 — APPROVED** [review]

**Review by rithiksharma100 — APPROVED** [review]

**Review by Chhekur — APPROVED** [review]

**Review by abhaymaniyar — APPROVED** [review]

**Pull request merged**